moped-editor/src/queries/subprojects.js

```diff
@@ -53,3 +57,14 @@ export const DELETE_PROJECT_SUBPROJECT = gql`
     }
   }
 `;
+
+export const CLEAR_PARENT_PROJECT = gql`
+  mutation ClearParentProject($projectId: Int!) {
+    update_moped_project(
+      where: { parent_project_id: { _eq: $projectId } }
+      _set: { parent_project_id: null }
+    ) {
+      affected_rows
+    }
+  }
+`;
```

Comment:
This makes total sense here, but I wondered if this could belong in the `PROJECT_ARCHIVE` mutation and look something like below. The differences that I see is that it would happen in the same API call as the archive which could help make sure this happens each and every time, and also it means less code. It does use [an alias](https://graphql.org/learn/queries/) in the mutation since there is a conflict in the `update_moped_project` field names in the mutation.

```
export const PROJECT_ARCHIVE = gql`
  mutation ArchiveMopedProject($projectId: Int!) {
    update_moped_project(
      where: { project_id: { _eq: $projectId } }
      _set: { is_deleted: true }
    ) {
      affected_rows
    }
    clear_parent_project: update_moped_project(
      where: { parent_project_id: { _eq: $projectId } }
      _set: { parent_project_id: null }
    ) {
      affected_rows
    }
  }
`;
```

Reply:
+1 for doing the subproject updates in the archive mutation!

Reply:
Agree!! I thought it maybe would be better to have them separate in case the subproject update failed it wouldn't rollback all the changes, but now that I think about it, thats not a good reason at all, and I'd rather have the rollback. I'm going to clean up this code and use your mutation @mddilley , thank you

Reply:
Awesome - just tested again and it works wonderfully! Going to add that ✅ now